src/worker/vm/imports.ts

```diff
@@ -2,14 +2,70 @@ import { BigQuery } from '@google-cloud/bigquery'
 import * as contrib from '@posthog/plugin-contrib'
 import * as AWS from 'aws-sdk'
 import crypto from 'crypto'
+import dns from 'dns'
 import * as genericPool from 'generic-pool'
-import fetch from 'node-fetch'
-import * as pg from 'pg'
+import nodeFetch from 'node-fetch'
+import nodePostgres from 'pg'
+import { parse } from 'pg-connection-string'
 import snowflake from 'snowflake-sdk'
+import url from 'url'
 import * as zlib from 'zlib'
 
 import { writeToFile } from './extensions/test-utils'
 
+// 93.184.216.34 = example.com
+const RESTRICTED_IPS = ['93.184.216.34']
+
+const isIpv6 = (str: string) =>
+    /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/.test(
+        str
+    )
+
+const validateHostOrUrl = async (hostOrUrl: any) => {
+    if (typeof hostOrUrl != 'string') {
+        throw new Error(`Invalid host/URL ${hostOrUrl}`)
+    }
+    if (hostOrUrl.includes('localhost') || isIpv6(hostOrUrl)) {
```

Comment:
Loopback IP addresses (starting with `127.`) should be banned too by default, in addition to `localhost` (which is a synonym for `127.0.01`.

BTW why disallow all IPv6?

Reply:
All IPs are actually blocked, not only IPv6. It's just that a few lines down checks for IPv4s. 

Also input came from here:

https://github.com/PostHog/plugin-server/issues/395#issuecomment-844417191

Reply:
Ah, "IPv4 addresses return a null host", that's a tiny bit hacky.
As was mentioned somewhere, for a malicious actor it's trivial to bypass this using any domain, while regular users could just get frustrated (especially with a vague error message), so I'd like to push back on this.